src/tests/Microsoft.Agents.Authentication.Msal.Tests/MsalAuthTests.cs

```diff
@@ -0,0 +1,260 @@
+
+            Assert.Equal(newToken, result);
+            Mock.Verify(options, service);
+        }
+
+        [Fact]
+        public async Task GetAccessTokenAsync_ShouldReturnTokenForClientCredentials()
+        {
+            var token = "token";
+
+            var options = new Mock<IOptions<MsalAuthConfigurationOptions>>();
+
+            var returnedOptions = new MsalAuthConfigurationOptions
+            {
+                MSALEnabledLogPII = false
+            };
+            options.Setup(x => x.Value).Returns(returnedOptions).Verifiable(Times.Exactly(2));
+
+            var logger = new Mock<ILogger<MsalAuth>>();
+
+            var service = new Mock<IServiceProvider>();
+            service.Setup(x => x.GetService(typeof(IOptions<MsalAuthConfigurationOptions>)))
+                .Returns(options.Object)
+                .Verifiable(Times.Exactly(2));
+            service.Setup(x => x.GetService(typeof(ILogger<MsalAuth>)))
+                .Returns(logger.Object)
+                .Verifiable(Times.Once);
+            service.Setup(sp => sp.GetService(typeof(IHttpClientFactory)))
+                .Returns(new TestHttpClientFactory())
+                .Verifiable(Times.Once);
+
+            var msalAuth = new MsalAuth(service.Object, _configuration.GetSection(SettingsSection));
+
+            var result = await msalAuth.GetAccessTokenAsync(ResourceUrl, _scopes, true);
+
+            Assert.Equal(token, result);
+            Mock.Verify(options, service);
+        }
+
+        [Fact]
+        public async Task GetAccessTokenAsync_ShouldReturnTokenForCertificate()
+        {
+            var token = "token";
+
+            var configSettings = new Dictionary<string, string> {
+                { "Connections:BotServiceConnection:Settings:AuthType", "Certificate" },
+                { "Connections:BotServiceConnection:Settings:ClientId", "test-id" },
+                { "Connections:BotServiceConnection:Settings:AuthorityEndpoint", "https://botframework/test.com" },
+                { "Connections:BotServiceConnection:Settings:CertThumbprint", "thumbprint" },
+                { "Connections:BotServiceConnection:Settings:Scopes:scope1", "{instance}" }
+            };
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(configSettings)
+                .Build();
+
+            var options = new Mock<IOptions<MsalAuthConfigurationOptions>>();
+
+            var returnedOptions = new MsalAuthConfigurationOptions
+            {
+                MSALEnabledLogPII = false
+            };
+            options.Setup(x => x.Value).Returns(returnedOptions).Verifiable(Times.Exactly(2));
+
+            var logger = new Mock<ILogger<MsalAuth>>();
+
+            var certificate = new Mock<ICertificateProvider>();
+            certificate.Setup(x => x.GetCertificate())
+                .Returns(CreateSelfSignedCertificate("test"))
+                .Verifiable(Times.Once); ;
```

Comment:
there are two `; ;` in multiple lines of this file

Reply:
Fixed. Thanks!